component/file_cache/file_cache.go

```diff
@@ -987,13 +987,10 @@ func (fc *FileCache) CloseFile(options internal.CloseFileOptions) error {
 
 	localPath := filepath.Join(fc.tmpPath, options.Handle.Path)
 
-	if options.Handle.Dirty() {
-		log.Info("FileCache::CloseFile : name=%s, handle=%d dirty. Flushing the file.", options.Handle.Path, options.Handle.ID)
-		err := fc.FlushFile(internal.FlushFileOptions{Handle: options.Handle}) //nolint
-		if err != nil {
-			log.Err("FileCache::CloseFile : failed to flush file %s", options.Handle.Path)
-			return err
-		}
+	err := fc.FlushFile(internal.FlushFileOptions{Handle: options.Handle}) //nolint
```

Comment:
In FileCache.CloseFile() shouldn't we call FlushFile() only when the handle is dirty?

Reply:
flush file internally checks the dirty flag, because flush can be called from outside also. As its already checked inside the method i have removed this.